lib/exchange.dart

```diff
@@ -387,6 +387,11 @@ class _ExchangeWidgetState extends State<ExchangeWidget> {
     showAlertDialog(context, 'creating order..');
     var res = await beOrderCreate(_market.symbol, _side, _amount);
     Navigator.pop(context);
+    _amountController.text = '';
+    _receiveController.text = '';
+    setState(() {
+      _validAmount = false;
+    });
     res.when(
         (order) => Navigator.push(
```

Comment:
you arent checking if the call was successful.

it needs to go in *this* block.

ie. **only** when the order was created successfully.